### PR TITLE
Enhance config validation for bad regex in CWAgent

### PR DESCRIPTION
**Pull request by taohungyang**
# Description of the issue
- If we put bad regex in the configuration json, currently the `amazon-cloudwatch-agent-ctl` will pass all the validation and start the agent. It seems to start the agent but we will later find in the log of CWagent errors like this:
  ```
  2022-04-25T12:18:37Z E! [agent] Service for [inputs.logfile] failed to start: filter regex has issue, regexp: Compile(^(?!NVRM:) ): error parsing regexp: invalid or unsupported Perl syntax: `(?!`
  2022-04-25T12:18:37Z E! [telegraf] Error running agent: filter regex has issue, regexp: Compile( ^(?!test:) ): error parsing regexp: invalid or unsupported Perl syntax: `(?!`
  ```
  We should error out before `amazon-cloudwatch-agent-ctl` ends and let the customer know the regex is bad.

- This PR deals with the configuration in:
  - LogFiltering
  - TimeStamp Format

# Description of changes
- Compile the regular expression in config translator when applying rules
- Add them in the translator errorMessages if it cannot compile. This will later be logged and panicked when translating json to toml.
- Update test cases

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
- make test

### CODE REVIEW

**Comment by sethAmazon** [comment]
I do not think this requires an integration test so lgtm.

**Review by sethAmazon — DISMISSED** [review]

File: translator/cmdutil/translatorutil.go
Comment (khanhntd) [inline translator/cmdutil/translatorutil.go:47]: I don't think you need to error out the message since [the executable for config-translator](https://github.com/aws/amazon-cloudwatch-agent/blob/master/cmd/config-translator/translator.go#L81-L83) has covered it

File: translator/translate/logs/logs_collected/files/collect_list/ruleLogFilters_test.go
Comment (khanhntd) [inline translator/translate/logs/logs_collected/files/collect_list/ruleLogFilters_test.go:77]: I don't think we need to assert retKey to begin with across the test since the final goal was to validate the errorMessage if having invalid reg_exp and the expect failed value

File: translator/cmdutil/translatorutil.go
Comment (taohungyang) [reply translator/cmdutil/translatorutil.go:47]: ohh..good catch. I read through that but forgot about it before modifying. Will update this in next commit

**Comment by khanhntd** [comment]
One final note for myself: Make sure to use `make fmt` before merging though since I saw some inconsistencies with `import` package

**Comment by taohungyang** [comment]
> One final note for reminding: Make sure to use `make fmt` before merging though since I saw some inconsistencies with `import` package

@khanhntd I just ran `make fmt` and see some 10+ files got formatted that I didn't update this time. Is that expected?

**Comment by khanhntd** [comment]
> > One final note for reminding: Make sure to use `make fmt` before merging though since I saw some inconsistencies with `import` package
> 
> @khanhntd I just ran `make fmt` and see some 10+ files got formatted that is not related to this change. Is that expected?

Yes. That's expected though. Mostly since other PR does not `make fmt` before merging

**Review by khanhntd — DISMISSED** [review]
LGTM.

**Review by SaxyPandaBear — APPROVED** [review]
lgtm

File: translator/translate/logs/logs_collected/files/collect_list/ruleTimestampFormat.go
Comment (SaxyPandaBear) [inline translator/translate/logs/logs_collected/files/collect_list/ruleTimestampFormat.go:165]: Is there a unit test for this? Sorry just caught it

Comment (taohungyang) [reply translator/translate/logs/logs_collected/files/collect_list/ruleTimestampFormat.go:165]: I did think about this before pushing out this PR but there was no `ruleTimestampFormat_test.go` at the moment. I can create one though

Comment (SaxyPandaBear) [reply translator/translate/logs/logs_collected/files/collect_list/ruleTimestampFormat.go:165]: That would be great. Gotta increase code coverage where it's easy 😃  otherwise LGTM so once you add that you can DM me and I'll approve if I don't see the email notification first

**Comment by taohungyang** [comment]
@SaxyPandaBear I found out that there are a lot of existing test cases for `ruleTimestampFormat` in [`collect_list_test.go`](https://github.com/aws/amazon-cloudwatch-agent/blob/master/translator/translate/logs/logs_collected/files/collect_list/collect_list_test.go#L62-L311)

While I do prefer separating those test cases that are not related to `collect_list.go` out to each individual `rule*_test.go`, I think it might be better to refactor this in another PR. Adding the new test case in this file for now.

File: translator/translate/logs/logs_collected/files/collect_list/collect_list_test.go
Comment (SaxyPandaBear) [inline translator/translate/logs/logs_collected/files/collect_list/collect_list_test.go:315]: I didn't notice that the name of this is misspelled. Looks like it was always like this. Not important to fix in this PR but a typo we should fix at some point

**Review by SaxyPandaBear — APPROVED** [review]
LGTM

**Pull request merged**
